main/test/ca/uwaterloo/flix/language/phase/TestResolver.scala

```diff
@@ -580,4 +580,44 @@ class TestResolver extends FunSuite with TestUtils {
     expectError[ResolutionError.UndefinedType](result)
   }
 
+  test("ImproperType.01") {
```

Comment:
- We usually name these tests the same as the error, e.g. IllegalUninhabitedType.01
- Could you add a few more tests cases (or maybe that is coming separately)?

Reply:
I've added some more cases covering different kinds and numbers of parameters. Let me know if there's an area I'm not covering sufficiently.